app/validators/questionnaire_schema.py

```diff
@@ -351,13 +373,25 @@ def _get_path_id(self, path):
     @lru_cache
     def get_block_id_by_answer_id(self, answer_id):
         for question, context in self.questions_with_context:
-            for answer in question.get("answers", []):
-                if answer_id == answer["id"]:
+            if dynamic_answers := question.get("dynamic_answers", {}):
+                if block_id := self.capture_block_id_for_answer(
+                    answer_id, context, dynamic_answers["answers"]
+                ):
+                    return block_id
+            if answers := question.get("answers", []):
+                if block_id := self.capture_block_id_for_answer(
+                    answer_id, context, answers
+                ):
+                    return block_id
```

Comment:
Based on previous comments, this can be simplified to:
```suggestion
            if block_id := self.capture_block_id_for_answer(
                answer_id, context, self.get_answers_from_question(question)
            ):
                return block_id
```

Reply:
Had to do this by hand because the method name has changed.